### PR TITLE
- updated method getCarById, use CarService interface as DI in carController; change name from Car -> CarEntity in all package and classes;

### PR DESCRIPTION
**Pull request by Miller1995**
- add new field in CarDTO, generated getter and setter;
- add in CarService interface new contract CarEntityToCarDTO, this contract get a parameter of type CarEntity;
- add new method in CarServiceImp, in this method built logic for mapping object from CarEntity to CarDTO;
- change getRequest in CarController, now the method getCarById() return a Http response of type ResponseEntity and because ResponseEntity is a generic type, we can use any type the response body;
- in CarController delete get request getCars();

### CODE REVIEW

File: src/main/java/md/miller1995/Dealership/models/entities/CarEntity.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
-public class Car {
+public class CarEntity {
     @Id
     @Column(name = "id")
```
Comment (AlexIliescu11) [inline src/main/java/md/miller1995/Dealership/models/entities/CarEntity.java:20]: Don't we need/want an autoincrement strategy for `id`? (@GeneratedValue(strategy=GenerationType.AUTO))
https://stackoverflow.com/questions/2011528/hibernate-auto-increment-id

File: src/main/java/md/miller1995/Dealership/services/CarServiceImp.java [hunk 2]
```diff
@@ -20,12 +21,23 @@ public CarServiceImp(CarRepository carRepository) {
...
 
-    public Car findById(long id){
-        Optional<Car> foundCar = carRepository.findById(id);
+    public CarEntity findById(long id){
+        Optional<CarEntity> foundCar = carRepository.findById(id);
         return foundCar.orElse(null);
     }
+
+    @Override
+    public CarDTO CarEntityToCarDTO(CarEntity carEntity) {
```
Comment (AlexIliescu11) [inline src/main/java/md/miller1995/Dealership/services/CarServiceImp.java:34]: Do we have a method name with an upper case letter?
Also, can't we maybe make use of ObjectMapper for entity -> dto conversion?

**Review by AlexIliescu11 — APPROVED** [review]

File: src/main/java/md/miller1995/Dealership/models/entities/CarEntity.java (hunk 1 above)
Comment (Miller1995) [reply src/main/java/md/miller1995/Dealership/models/entities/CarEntity.java:17]: - add annotation @GeneratedValue for field id in CarEntity class;

File: src/main/java/md/miller1995/Dealership/services/CarServiceImp.java (hunk 2 above)
Comment (Miller1995) [reply src/main/java/md/miller1995/Dealership/services/CarServiceImp.java]: - changed name of method with lower case letter;
- use modelMapper for conversion entity -> dto

**Pull request merged**